spec/models/knight_spec.rb

```diff
@@ -1,5 +1,34 @@
 require 'rails_helper'
 
 RSpec.describe Knight, type: :model do
-  
+
+  describe "#valid_move?" do
+
+      {
+        #starting at edge of board
+        [[0, 0], [1, 2]] => true,
+        [[0, 0], [1, -2]] => false, # off-board
+        [[0, 0], [2, 1]] => true, 
+        [[0, 0], [2, -1]] => false, # off-board
+        [[0, 0], [0, 0]] => false, # no move, wrong logic
+        
+        #starting at center of board
+        [[3, 3], [1, 2]] => true,
+        [[3, 3], [1, 4]] => true,
+        [[3, 3], [5, 2]] => true,
+        [[3, 3], [5, 4]] => true,
+        [[3, 3], [4, 1]] => true,
+        [[3, 3], [4, 5]] => true,
+        [[3, 3], [2, 1]] => true,
+        [[3, 3], [2, 5]] => true,
+        [[3, 3], [1, 1]] => false, # wrong logic
+        [[3, 3], [0, 0]] => false, # wrong logic
+        [[3, 3], [-1, -1]] => false, # wrong logic & off-board
+      }.each do |before_and_after_coords, result|
+        it "returns #{result} when moving from #{before_and_after_coords.first} to #{before_and_after_coords.last}" do
+          knight = Knight.create(x_cord: before_and_after_coords.first.first, y_cord: before_and_after_coords.first.last)
+          expect(knight.valid_move?(before_and_after_coords.last.first, before_and_after_coords.last.last)).to eq result
+        end
+      end
```

Comment:
Oh my gosh, you have no idea... we started with like 5+ tests fully fledged
and kept thinking of scenarios to test, so finally redid the entire thing
this way instead.

On Tue, Jul 5, 2016 at 3:54 PM, Aleks Real <notifications@github.com> wrote:

> In spec/models/knight_spec.rb
> <https://github.com/Methodologists/metchess/pull/20#discussion_r69636123>:
>
> > +        [[3, 3], [1, 4]] => true,
> > +        [[3, 3], [5, 2]] => true,
> > +        [[3, 3], [5, 4]] => true,
> > +        [[3, 3], [4, 1]] => true,
> > +        [[3, 3], [4, 5]] => true,
> > +        [[3, 3], [2, 1]] => true,
> > +        [[3, 3], [2, 5]] => true,
> > +        [[3, 3], [1, 1]] => false, # wrong logic
> > +        [[3, 3], [0, 0]] => false, # wrong logic
> > +        [[3, 3], [-1, -1]] => false, # wrong logic & off-board
> > +      }.each do |before_and_after_coords, result|
> > +        it "returns #{result} when moving from #{before_and_after_coords.first} to #{before_and_after_coords.last}" do
> > +          knight = Knight.create(x_cord: before_and_after_coords.first.first, y_cord: before_and_after_coords.first.last)
> > +          expect(knight.valid_move?(before_and_after_coords.last.first, before_and_after_coords.last.last)).to eq result
> > +        end
> > +      end
>
> These tests are great! I'm taking notes 👍
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Methodologists/metchess/pull/20/files/d7c0e7ba6e344ed2066706f3544af530ea19a433#r69636123>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ALOK7EJy5iXybEPpmgZlsT41tjskdxWGks5qSsR2gaJpZM4JDs8Y>
> .
>



-- 
-Lin Rui Li